contracts/PositionManagerOngoingInterest.sol

```diff
@@ -0,0 +1,170 @@
+    {
+        _payInterest();
+        super.managePosition(
+            collateralChange, isCollateralIncrease, debtChange, isDebtIncrease, maxFeePercentage, permitSignature
+        );
+    }
+
+    error NotSupported();
+
+    function redeemCollateral(uint256, uint256, ERC20PermitSignature calldata) public override {
+        /// TODO: IMPORTANT wut do? revert NotSupported()? or add support
+        revert NotSupported();
+    }
+
+    /// TODO: IMPORTANT create interface and add dis?
+    function liquidate(address position) external unlockCollateralToken {
+        uint256 positionDebt = debtToken.balanceOf(position);
+
+        _rToken.transferFrom(msg.sender, address(this), positionDebt);
+
+        uint256 balBefore = wrappedCollateralToken.balanceOf(address(this));
+        IPositionManager(positionManager).liquidate(position);
+        uint256 liquidatorReward = wrappedCollateralToken.balanceOf(address(this)) - balBefore;
+        wrappedCollateralToken.withdrawTo(msg.sender, liquidatorReward);
+
+        /// TODO: IMPORTANT discuss
+        pendingInterestStored[position] = 0;
+        lastPendingInterestUpdate[position] = block.timestamp;
+    }
+
+    function _payInterest() internal {
+        uint256 pendingInterest_ = pendingInterest(msg.sender);
+        lastPendingInterestUpdate[msg.sender] = block.timestamp;
+        if (pendingInterest_ == 0) return;
+
+        uint256 maxMintableDebt = _getMaxMintableDebt(msg.sender);
+        uint256 debtToMint;
+        uint256 appliedBorrowingFee;
+        if (pendingInterest_ > maxMintableDebt) {
+            (debtToMint, appliedBorrowingFee) = _applyBorrowingFee(maxMintableDebt);
+            pendingInterestStored[msg.sender] = pendingInterest_ - maxMintableDebt;
+        } else {
+            (debtToMint, appliedBorrowingFee) = _applyBorrowingFee(pendingInterest_);
+            pendingInterestStored[msg.sender] = 0;
+        }
+
+        ERC20PermitSignature memory emptySignature;
+        (, uint256 actualDebtChange) = IPositionManager(positionManager).managePosition(
+            wrappedCollateralToken, msg.sender, 0, false, debtToMint, true, appliedBorrowingFee, emptySignature
+        );
+
+        emit InterestPaid(msg.sender, actualDebtChange, pendingInterestStored[msg.sender]);
+    }
+
+    // IPositionManager(positionManager).splitLiquidationCollateral(wrappedCollateralToken).MCR()
+    function _getMaxMintableDebt(address position) internal returns (uint256) {
+        (uint256 price,) = priceFeed.fetchPrice();
+        uint256 collateralBalance = raftCollateralToken.balanceOf(position);
+        uint256 debtBalance = debtToken.balanceOf(position);
+        uint256 MCR = IPositionManager(positionManager).splitLiquidationCollateral(wrappedCollateralToken).MCR();
+
+        uint256 maxDebt = collateralBalance * price / MCR;
+        return debtBalance >= maxDebt ? 0 : maxDebt - debtBalance;
+    }
+
+    function _applyBorrowingFee(uint256 amount)
```

Comment:
Alternatively, we can put the lock on pricefeed and disable redemptions completely. Then we can assume borrowing fee is always 0.